e2e-tests/upgrade-haproxy/run

```diff
@@ -43,11 +42,6 @@ IMAGE_PROXY=$(echo "${INIT_OPERATOR_IMAGES}" | jq -r '.versions[].matrix.proxysq
 IMAGE_HAPROXY=$(echo "${INIT_OPERATOR_IMAGES}" | jq -r '.versions[].matrix.haproxy[].imagePath')
 IMAGE_BACKUP=$(echo "${INIT_OPERATOR_IMAGES}" | jq -r '.versions[].matrix.backup[].imagePath')
 
-if [[ ${TARGET_API} == "${API}" ]]; then
```

Comment:
Let's not remove this, but instead compare TARGET_OPERATOR_VER and INIT_OPERATOR_VER. This is a safety mechanism to know that the test is actually doing something meaningful.

Reply:
nice! I think it will better!

Reply:
O! I remember I didn't do like this 
we have condition 
`if [[ ${INIT_OPERATOR_VER} == "${TARGET_OPERATOR_VER}" ]]; then
	INIT_OPERATOR_VER=$(curl -s https://check.percona.com/versions/v1/pxc-operator | jq -r '.versions[].operator' | sort -V | tail -n2 | head -n1)
fi`
so it's impossible to have the same versions

Reply:
It is unlikely, but not impossible, here's why: if `if [[ ${INIT_OPERATOR_VER} == "${TARGET_OPERATOR_VER}" ]];` returns the same version we run the next command to reset `INIT_OPERATOR_VER` but this command may have error for some reason (eg. version service can have a bug or if we added some faulty json to it) and still return the same version so I prefer to leave the safety check since for upgrade test it is critical that we don't use the same version (and it's a quick check also).

Reply:
ok! np I will add it.